sources/packages/web/src/store/modules/student/getters.ts

```diff
@@ -1,7 +1,23 @@
 import { GetterTree } from "vuex";
-import { RootState } from "@/types";
+import {
+  RootState,
+  RestrictionNotificationType,
+  StudentRestriction,
+} from "@/types";
 import { StudentState } from "./student";
 
 export const getters: GetterTree<StudentState, RootState> = {
-  // To be implemented if needed.
+  hasRestriction(state: StudentState): boolean {
+    return state.restrictions?.some(
```

Comment:
We can ensure that the state will always have an empty array and avoid the null checks every time.